blocks/subquery/edit.tsx

```diff
@@ -208,22 +223,48 @@ export default function Edit({
               context: 'edit',
             },
           ),
-        }).then((response) => (response as any as WpRestApiPosts).map((post) => post.id));
+        }).then((result) => {
+          const resultIds = result.map((post) => post.id);
+          setAttributes({ validPosts: resultIds });
+        });
```

Comment:
With `await` I wouldn't use `then`, I'd set the return value. I'd also not trust that the return from the API fetch is what we think it is. Something like:
```ts
const response = await apiFetch<unknown>({ /* etc */ });
const resultIds: number[] = Array.isArray(response)
  ? response.map((post: unknown) => Number.isInteger(post?.id) ? post.id : 0).filter()
  : [];
setAttributes({ validPosts: resultIds });
```

Reply:
Makes sense! I think I was getting tripped up in my brain because it wasn't an 'outside' API, but that doesn't really matter. Ty for the explanation.